server.go

```diff
@@ -242,10 +242,20 @@ func (s *Server) handleNextGame(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	gh, ok := s.games[request.GameID]
-	if !ok || request.CreateNew {
+	if !ok {
+		// new random state game if no game exists
 		gh = newHandle(newGame(request.GameID, randomState(words)), s.Store)
-		s.games[request.GameID] = gh
+	} else {
+		state, ok := decodeGameState(gh.g.GameState.ID(), s.defaultWords)
```

Comment:
No need to decode the game state here. A bit of background: the game state ID was added for reconstructing games after a process start. Previously games weren't saved to disk anywhere, so if the process restarted we recreated the game from the state ID that the client sent up in its polling loop.

Currently, games are saved to a local kv database, and we can get rid of the decodeGameState and GameState.ID logic.

So specifically here I think we can just make this `else` branch an `else if request.CreateNew {` and in the body use `gh.g.GameState` to construct a new game.

Reply:
Way nicer. Thanks. Done